astroid/nodes/node_classes.py

```diff
@@ -955,7 +991,9 @@ def find_argname(self, argname, rec=DEPRECATED_ARGUMENT_DEFAULT):
                 stacklevel=2,
             )
         if self.arguments:
-            return _find_arg(argname, self.arguments)
+            index, argument = _find_arg(argname, self.arguments)
+            if argument and argument.name not in [self.vararg, self.kwarg]:
+                return index, argument
```

Comment:
In a follow-up issue we could look into whether we should be able to return the names of `*args` and `**kwargs`.